api/turing/cluster/controller.go

```diff
@@ -488,7 +489,23 @@ func (c *controller) waitKnativeServiceReady(
 	for {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("timeout waiting for service %s to be ready", svcName)
+			terminationMessage := c.getKnativePodTerminationMessage(svcName, namespace)
+			if terminationMessage == "" {
+				// Pod was not created (as with invalid image names), get most recent status message
+				// from the knative service
+				svc, err := services.Get(svcName, metav1.GetOptions{})
+				if err != nil {
+					terminationMessage = err.Error()
+				} else {
+					failedCondition := svc.Status.GetCondition(knativeapis.ConditionType("ConfigurationsReady"))
+					if failedCondition != nil && failedCondition.IsFalse() {
+						terminationMessage = failedCondition.GetMessage()
```

Comment:
Maybe we can use this const for the "ConfigurationsReady" condition type?
https://github.com/knative/serving/blob/7398ed7b7e45bb751e6aa475a8d2f2ab15398e44/pkg/apis/serving/v1/service_types.go#L101

I think the variable name `failedCondition` can be misinterpreted, one would think `failedCondition.IsFalse()` means condition is successful hence the configuration is Ready, when it is actually not.

Can we maybe use `configurationReadyCondition` or something similar as the variable name?